build-tools/packages/build-cli/src/lib/changesets.ts

```diff
@@ -59,7 +59,9 @@ export async function loadChangesets(dir: string, log?: Logger): Promise<Changes
 		// Get the date the changeset file was added to git.
 		// eslint-disable-next-line no-await-in-loop
 		const results = await repo.gitClient.log({ file, maxCount: 1, strictDate: true });
-		const added = parseISO(results.all[0].date);
+		
+		// Newly added files won't have any results from git log, so default to now.
+		const added = parseISO(results.all?.[0]?.date ?? formatISO(Date.now()));
```

Comment:
I was torn between using native `Date.now().toISOString()` or sticking to the library. No strong feelings one way or the other, so feel free to merge like this.

Reply:
I had the same internal wrestling match. :smile: I went with this for symmetry, but it probably doesn't matter except for some unknown edge cases we probably won't hit.